py/src/braintrust/cli/push.py

```diff
@@ -0,0 +1,146 @@
+"""Implements the braintrust push subcommand."""
+import importlib.util
+import inspect
+import os
+import subprocess
+import tempfile
+import zipfile
+
+import requests
+
+from .. import api_conn, app_conn, login, org_id, proxy_conn
+from ..framework2 import global_
+
+
+def run(args):
+    """Runs the braintrust push subcommand."""
+    login(
+        api_key=args.api_key,
+        org_name=args.org_name,
+        app_url=args.app_url,
+    )
+
+    # Execute the user's file as a module.
+    spec = importlib.util.spec_from_file_location("unused", args.file)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+
+    # Fetch a bundle ID to which to upload.
+    resp = proxy_conn().post_json(
+        "function/code",
+        {
+            "org_id": org_id(),
+            "runtime_context": {
+                "runtime": "python",
+                "version": "3.12",
```

Comment:
in JS i think we dynamically infer this and support a fixed set of versions (which we reject in the `/function/code` API). Should we do the same for python?

Reply:
Done